### PR TITLE
Fixed #26009 -- Fixed contenttypes_tests isolation.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/26009

### CODE REVIEW

File: tests/contenttypes_tests/test_order_with_respect_to.py [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 from .models import Answer, Post, Question
 
 
-class OrderWithRespectToGFKTests(OrderWithRespectToTests):
+class OrderWithRespectToGFKTests(BaseOrderWithRespectToTests):
     Answer = Answer
     Question = Question
 
-del OrderWithRespectToTests
+del BaseOrderWithRespectToTests
```
Comment (timgraham) [inline tests/contenttypes_tests/test_order_with_respect_to.py]: Alternately to this pattern, the base classes could inherit `object` (so the test runner doesn't pick them up) and the subclasses could inherit the base class as a mixin as well as `TestCase`. This is slightly repetitive though, so I couldn't quite decide which is better.

Comment (charettes) [inline tests/contenttypes_tests/test_order_with_respect_to.py]: I think the mixin and `TestCase` approach is more common in the tests code base.

Comment (timgraham) [inline tests/contenttypes_tests/test_order_with_respect_to.py]: Updated to use that approach.

**Comment by charettes** [comment]
LGTM

**Pull request merged**